scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,835 @@
+                                   "Run conditionalReplaceAll" + ON_KEY + QUOTE + currentKey + QUOTE + RESULT_IS +
+                                            answer,
+                                   Level.DEBUG);
+                }
+            }
+            commitIfNeeded(channel);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public void conditionalReplaceAll(String originator, TaskId taskid, String channel, String predicate,
+            String thenRemappingFunction, String elseRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
+        String currentKey = null;
+        try {
+            Channel<Serializable> chosenChannel = getChannel(channel);
+            for (Map.Entry<String, Serializable> entry : chosenChannel.entrySet()) {
+                currentKey = entry.getKey();
+                Serializable value = chosenChannel.get(currentKey);
+                if (evaluateClosure(predicate, BiPredicate.class).test(currentKey, value)) {
+
+                    Serializable answer = chosenChannel.compute(currentKey,
+                                                                (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(thenRemappingFunction,
+                                                                                                                                                           BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll(Then)" + ON_KEY + QUOTE + currentKey + QUOTE + RESULT_IS +
+                                            answer,
+                                   Level.DEBUG);
+                } else {
+                    Serializable answer = chosenChannel.compute(currentKey,
+                                                                (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(elseRemappingFunction,
+                                                                                                                                                           BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll(Else)" + ON_KEY + QUOTE + currentKey + QUOTE + RESULT_IS +
+                                            answer,
+                                   Level.DEBUG);
+
+                }
+            }
+            commitIfNeeded(channel);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)